cc/sgx/services/fns/sgx_trusted_lib/src/trusted_worker/operators.rs

```diff
@@ -0,0 +1,315 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// Insert std prelude in the top for the sgx feature
+#[cfg(feature = "mesalock_sgx")]
+use std::prelude::v1::*;
+use std::vec;
+use std::iter::FromIterator;
+use std::convert::TryInto;
+
+use crate::worker::{Worker, WorkerContext};
+use crate::trusted_worker::register_func;
+use eigen_core::{Error, ErrorKind, Result};
+use crate::eigen_crypto::sign::ecdsa::KeyPair;
+
+// TODO: unwrap() method should all be replaced with a property error handling
+
+pub struct OperatorWorker {
+    worker_id: u32,
+    func_name: String,
+    input: Option<OperatorWorkerInput>,
+}
+
+impl OperatorWorker {
+    pub fn new() -> Self {
+        OperatorWorker {
+            worker_id: 0,
+            func_name: "operator".to_string(),
+            input: None,
+        }
+    }
+}
+
+enum OperatorKind {
+    Add,
+    Add1,
+    Sub,
+    Sub1,
+    Enc,
+    Des,
+}
+
+struct OperatorWorkerInput {
+    // TODO: Use a more property field name to describe what we want to save here.
+    //       e.g., when operator is `Enc`, `cipher_op1` is not a cipher one
+    op: OperatorKind,
+    cipher_op1: String,
+    cipher_op2: String,
+}
+
+impl Worker for OperatorWorker {
+    fn function_name(&self) -> &str {
+        self.func_name.as_str()
+    }
+
+    fn set_id(&mut self, worker_id: u32) {
+        self.worker_id = worker_id;
+    }
+
+    fn id(&self) -> u32 {
+        self.worker_id
+    }
+
+    fn prepare_input(&mut self, dynamic_input: Option<String>) -> Result<()> {
+        let msg = dynamic_input.ok_or_else(|| Error::from(ErrorKind::InvalidInputError))?;
+
+        // `args` should be "op,op_num,op1,op2,op3,..."
+        // now `op` may be 'add' or 'sub'
+        // number is parsed with big endian
+
+        let splited = Vec::from_iter(msg.split(",").map(String::from));
+
+        let op = splited.get(0).unwrap();
+
+        match op.as_str() {
+            "add" => {
+                let op_num = splited.get(1).unwrap().parse::<u64>().unwrap();
```

Comment:
`let vec: Vec<&str> = msg.split(",").collect(); `  then you can use index to access the value after the arguments number check.